app/src/main/java/info/nightscout/androidaps/plugins/configBuilder/ProfileFunctionImplementation.kt

```diff
@@ -43,7 +43,7 @@ class ProfileFunctionImplementation @Inject constructor(
     private val deviceStatusData: DeviceStatusData
 ) : ProfileFunction {
 
-    val cache = LongSparseArray<Profile>()
+    val cache = HashMap<Long,Profile?>()
```

Comment:
here we do care about speed much more. so few extra kb of memory are fair price imo :)
but surely it can be changed back if you think it is critical.